android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -1558,19 +1560,25 @@ private void clearWalletModelServices() {
     }
 
     private void setupWalletModel() {
-        if (mWalletModel == null) {
-            mWalletModel = new WalletModel(getApplicationContext(), mKeyringService,
-                    mBlockchainRegistry, mJsonRpcService, mTxService, mEthTxManagerProxy,
-                    mSolanaTxManagerProxy, mAssetRatioService, mBraveWalletService, mSwapService);
-        } else {
-            mWalletModel.resetServices(getApplicationContext(), mKeyringService,
-                    mBlockchainRegistry, mJsonRpcService, mTxService, mEthTxManagerProxy,
-                    mSolanaTxManagerProxy, mAssetRatioService, mBraveWalletService, mSwapService);
-        }
-        setupObservers();
+        PostTask.postTask(TaskTraits.UI_DEFAULT, () -> {
```

Comment:
I'm a bit confused about what is going on here, what thread is this called from?